configserver/src/test/java/com/yahoo/vespa/config/server/deploy/RedeployTest.java

```diff
@@ -55,5 +65,50 @@ public void testNoRedeploy() {
                                               InstanceName.from("default"));
         assertFalse(tester.redeployFromLocalActive(id).isPresent());
     }
-    
+
+    @Test
+    public void testRedeployWillPurgeOldNonActiveDeployments() {
+        ManualClock clock = new ManualClock(Instant.now());
+        ConfigserverConfig configserverConfig = new ConfigserverConfig(new ConfigserverConfig.Builder()
+                                                                               .configServerDBDir(Files.createTempDir()
+                                                                                                       .getAbsolutePath())
+                                                                               .sessionLifetime(60));
+        DeployTester tester = new DeployTester("src/test/apps/app", configserverConfig, clock);
+        tester.deployApp("myapp", Instant.now()); // session 2 (numbering starts at 2)
+
+        clock.advance(Duration.ofSeconds(10));
+        Optional<com.yahoo.config.provision.Deployment> deployment2 = tester.redeployFromLocalActive();
+
+        assertTrue(deployment2.isPresent());
+        deployment2.get().activate(); // session 3
+        long activeSessionId = tester.tenant().getLocalSessionRepo().getActiveSession(tester.applicationId()).getSessionId();
+
+        clock.advance(Duration.ofSeconds(10));
+        Optional<com.yahoo.config.provision.Deployment> deployment3 = tester.redeployFromLocalActive();
+        assertTrue(deployment3.isPresent());
+        deployment3.get().prepare();  // session 4 (not activated)
+
+        try { Thread.sleep(1000);} catch (InterruptedException e) { /* Ignored */ }
```

Comment:
Why do we need to sleep 1 second?

Reply:
We don't have to, just added while writing the test, will remove